src/cpp/session/modules/rmarkdown/SessionRMarkdown.cpp

```diff
@@ -1305,8 +1306,57 @@ Error maybeCopyWebsiteAsset(const json::JsonRpcRequest& request,
    return Success();
 }
 
-
-
+Error executeAlternateEngineChunk(const json::JsonRpcRequest& request,
+                                  json::JsonRpcResponse* pResponse)
+{
+   std::string docId, chunkId, engine, code;
+   Error error = json::readParams(request.params,
+                                  &docId,
+                                  &chunkId,
+                                  &engine,
+                                  &code);
+   if (error)
+   {
+      LOG_ERROR(error);
+      return error;
+   }
+   
+   if (string_utils::toLower(engine) == "python")
```

Comment:
I plan to refactor / clean this up soon, but I imagine we will need a bit of engine specific code (eg when using the Rscript engine, we might need to invoke `Rscript -f` or something to that effect. 

Either way,  I definitely don't plan on branching for each engine we support once the PR is ready; it'll be tidier than that :)